### PR TITLE
fix(angular-apollo-tailwind): append protocol if missing from user url

### PR DESCRIPTION
**Pull request by asincole**
Refs: #591

### CODE REVIEW

File: angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html [hunk 1]
```diff
@@ -74,7 +74,7 @@ <h1 class="mt-2">
           ></hero-icon>
           <a
             class="link"
-            [href]="user.websiteUrl"
+            [href]="generateValidUrl(user.websiteUrl)"
```
Comment (ktrz) [inline angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html:77]: Could you please implement this as pipe? Calling functions in template is not good for performance

**Review by ktrz — DISMISSED** [review]

File: angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html (hunk 1 above)
Comment (asincole) [reply angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html:77]: oh sure.

Comment (asincole) [reply angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html]: done @ktrz

File: angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html [hunk 2]
```diff
@@ -88,7 +88,7 @@ <h1 class="mt-2">
           ></sd-twitter-icon>
           <a
             class="link"
-            [href]="'https:/twitter.com/' + user.twitterUsername"
+            [href]="'https://twitter.com/' + user.twitterUsername"
```
[the remark this answers is not in the document]
Comment (asincole) [reply angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html:91]: there is already a check that only displays this if its not null or undefined a few lines above it. as to your suggestion 

@ktrz what do you think about this bit?

[the remark this answers is not in the document]
Comment (asincole) [reply angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html:91]: @ihardz look at this comment above https://github.com/thisdot/starter.dev-github-showcases/pull/637#discussion_r972992991 on this

[the remark this answers is not in the document]
Comment (ktrz) [reply angular-apollo-tailwind/src/app/profile/profile-about/profile-about.component.html:91]: Moving this logic out of the component is definitely a great idea but we shouldn't call functions from a component in the template for performance reasons so pipe seems like a better place. This way you could also add some basic unit tests easily.

Feel free to do it as part of this PR or just create a new issue + PR

File: angular-apollo-tailwind/projects/shared/src/lib/pipes/generate-url-with-protocol/generate-url-with-protocol.pipe.spec.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { GenerateUrlWithProtocolPipe } from './generate-url-with-protocol.pipe';
+
+describe('GenerateUrlWithProtocolPipe', () => {
+  it('create an instance', () => {
+    const pipe = new GenerateUrlWithProtocolPipe();
+    expect(pipe).toBeTruthy();
+  });
```
Comment (ktrz) [inline angular-apollo-tailwind/projects/shared/src/lib/pipes/generate-url-with-protocol/generate-url-with-protocol.pipe.spec.ts:7]: Let's add couple of basic unit tests for that logic :)
Unit tests should be part of each new PR. Especially if we're fixing a bug

**Review by lindakatcodes — COMMENTED** [review]
Think this looks reasonable to me, but would be good to have Chris or Ihar re-review it as well! I didn't have a ton of time left this week to thoroughly test this, but it seems like everything is in place.

**Review by lindakatcodes — APPROVED** [review]
Tested again locally and appears to be working as expected!

**Pull request merged**